sqlx-core/src/any/row.rs

```diff
@@ -120,8 +120,8 @@ impl AnyRow {
                 _ if value.is_null() => AnyValueKind::Null,
                 AnyTypeInfoKind::Null => AnyValueKind::Null,
                 AnyTypeInfoKind::Bool => AnyValueKind::Bool(decode(value)?),
-                AnyTypeInfoKind::SmallInt => AnyValueKind::SmallInt(decode(value)?),
-                AnyTypeInfoKind::Integer => AnyValueKind::Integer(decode(value)?),
+                AnyTypeInfoKind::SmallInt => AnyValueKind::BigInt(decode(value)?),
+                AnyTypeInfoKind::Integer => AnyValueKind::BigInt(decode(value)?),
                 AnyTypeInfoKind::BigInt => AnyValueKind::BigInt(decode(value)?),
```

Comment:
Just noticed I put this somewhere non-sqlite specific, have to find a spot where it only affects SQLite.